### ISSUE TITLE
Remember last maintenance time

### ISSUE THREAD

**Issue by timuthy** [gardener/gardener#2097]
**What would you like to be added**:
Add a timestamp of the last triggered maintenance to the shoot `status`. The maintenance controller should consider this timestamp for maintenance decisions.

**Why is this needed**:
Gardener executes different operations like infrastructure reconciliation, version updates or shoot reconciliation (config dependent) only during the shoot's maintenance time (`shoot.spec.maintenance.timeWindow`) to avoid any possible disruption outside of this time window. 

The `Gardener-Controller-Manager` only checks if the shoot is currently in this maintenance time window but doesn't consider the last time the maintenance was triggered. So if the manager is restarted the maintenance might be triggered again.

**Comment by rfranzke** [comment]
Hm, another maintenance operation should be idempotent modulo the newly (to-be-)introduced restart of the pods. Is the effort of remembering the last maintenance really worth the benefit of avoiding restarts?

I can think of more benefits for end-users (which balance out the question). For example, users would be able to see when the last maintenance was and when the next maintenance is planned. This goes well with the potential issue of GCM being down during a shoot maintenance time window, skipping it entirely when it comes up only after it. 
A related issue is #1278. We could have similar timestamps for hibernation to get this one fixed in the same way.

WDYT?

**Comment by timuthy** [comment]
Let's pick this up for the next quality week.

**Comment by rfranzke** [comment]
TBH, I think there are more "pressing" topics that we should spent time with, WDYT? This is more a nice-to-have improvement, or?

**Comment by timuthy** [comment]
Sure, no objections.

/nice-to-have

**Comment by rfranzke** [comment]
This issue is open since a long time without anyone picking it up or any attention at all. Does everybody agree to closing it for now since the benefits seem to be not too important valuable for now?

**Comment by rfranzke** [comment]
I just noticed #3242 which might be related for considerations.

**Comment by timuthy** [comment]
> I just noticed #3242 which might be related for considerations.

Hm, is it related? This issues is more about GCM triggering the maintenance.
However, no objection against closing this now because of missing attention and dev capacity.
